extension/data_loader/file.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#ifndef _WIN32
+#include <unistd.h>
+#else
+#include <executorch/runtime/platform/compiler.h> // For ssize_t.
+#include <io.h>
+
+#include <windows.h>
+
+inline ssize_t pread(int __fd, void* __buf, size_t __nbytes, size_t __offset) {
+  OVERLAPPED overlapped; /* The offset for ReadFile. */
+  memset(&overlapped, 0, sizeof(overlapped));
+  overlapped.Offset = __offset;
+  overlapped.OffsetHigh = __offset >> 32;
```

Comment:
Will this do the right thing on both 32-bit and 64-bit architectures? Seems like it should, but I don't know enough about the definition of DWORD on different architectures

Reply:
DWORD will always be 32 bits on both 32-bit and 64-bit Windows systems.

It's important to note that the nNumberOfBytesToRead parameter for ReadFile is a DWORD. Therefore, if we need to read more than 4GB of data at once, the current implementation needs to be updated to handle larger file sizes.